src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -242,6 +242,7 @@ public override INamedTypeSymbol CreateErrorTypeSymbol(INamespaceOrTypeSymbol co
             bool isSubmission)
         {
             Debug.Assert(options != null);
+            Debug.Assert(!isSubmission || options.ReferencesSupersedeLowerVersions);
```

Comment:
I missed the fact that the option was internal. In that case, the current assert is reasonable.